ui/app/AppLayouts/Chat/components/PrivateChatPopupExistingContacts.qml

```diff
@@ -0,0 +1,63 @@
+import QtQuick 2.13
+import QtQuick.Controls 2.13
+import QtQuick.Layouts 1.13
+import "../../../../imports"
+import "../../../../shared"
+import "../../../../shared/status"
+import "./"
+
+Rectangle {
+    id: root
+    anchors.left: parent.left
+    anchors.right: parent.right
+    property string filterText: ""
+    property bool expanded: true
+    signal contactClicked(var contact)
+
+    function matchesAlias(name, filter) {
+        let parts = name.split(" ")
+        return parts.some(p => p.startsWith(filter))
+    }
+
+    height: Math.min(contactListView.contentHeight, (expanded ? 320 : 192))
+    ScrollView {
+        anchors.fill: parent
+        Layout.fillWidth: true
+        Layout.fillHeight: true
+
+        ScrollBar.horizontal.policy: ScrollBar.AlwaysOff
+        ScrollBar.vertical.policy: contactListView.contentHeight > contactListView.height ? ScrollBar.AlwaysOn : ScrollBar.AlwaysOff
+
+        ListView {
+            anchors.fill: parent
+            spacing: 0
+            clip: true
+            id: contactListView
+            model: profileModel.contacts.list
+            delegate: Contact {
+                showCheckbox: false
+                pubKey: model.pubKey
+                isContact: model.isContact
+                isUser: false
+                name: model.name
+                address: model.address
+                identicon: model.thumbnailImage || model.identicon
+                visible: model.isContact && (root.filterText === "" ||
+                    root.matchesAlias(model.name.toLowerCase(), root.filterText.toLowerCase()) ||
+                    model.name.toLowerCase().includes(root.filterText.toLowerCase()) ||
+                    model.address.toLowerCase().includes(root.filterText.toLowerCase()))
+                MouseArea {
```

Comment:
`Contact` already has a `MouseArea`, but I see why you added your here. maybe we should make an `onClicked` `var` prop on `Contact` that if it is defined, it calls that prop and not the "default" behaviour. What do you think?

Reply:
Yea, certainly an option. I personally think it's good when component only signal stuff, but we can make it with default logic.